File: src/com/Torvald/Terrarum/ConsoleCommand/CommandDict.java
package com.Torvald.Terrarum.ConsoleCommand;

import java.util.Hashtable;

/**
 * Created by minjaesong on 16-01-15.
 */
public class CommandDict {

    protected static Hashtable<String, ConsoleCommand> dict;

    public CommandDict() {
        dict = new Hashtable<>();

        dict.put("setav", new SetAV());
        dict.put("qqq", new QuitApp());
        dict.put("codex", new CodexEdictis());
        dict.put("export", new ExportMap());
        dict.put("gc", new ForceGC());
        dict.put("getav", new GetAV());
        dict.put("getlocale", new GetLocale());
        dict.put("togglenoclip", new ToggleNoClip());
        dict.put("nc", dict.get("togglenoclip"));
        dict.put("bulletintest", new SetBulletin());
        dict.put("setlocale", new SetLocale());
        dict.put("zoom", new Zoom());
        dict.put("teleport", new TeleportPlayer());
        dict.put("cat", new CatStdout());
        dict.put("exportav", new ExportAV());
        dict.put("gsontest", new GsonTest());
    }

    public static ConsoleCommand getCommand(String commandName) {
        return dict.get(commandName);
    }

}

File: src/com/Torvald/Terrarum/ConsoleCommand/CommandDict.kt
package com.Torvald.Terrarum.ConsoleCommand

import com.Torvald.Terrarum.Terrarum

import java.util.HashMap

/**
 * Created by minjaesong on 16-01-15.
 */
object CommandDict {

    internal var dict: HashMap<String, ConsoleCommand> = hashMapOf(
            Pair("echo", Echo()),
            Pair("setav", SetAV()),
            Pair("qqq", QuitApp()),
            Pair("codex", CodexEdictis()),
            Pair("export", ExportMap()),
            Pair("gc", ForceGC()),
            Pair("getav", GetAV()),
            Pair("getlocale", GetLocale()),
            Pair("togglenoclip", ToggleNoClip()),
            Pair("nc", ToggleNoClip()),
            Pair("setlocale", SetLocale()),
            Pair("zoom", Zoom()),
            Pair("teleport", TeleportPlayer()),
            Pair("tp", TeleportPlayer()),
            Pair("cat", CatStdout()),
            Pair("exportav", ExportAV()),
            Pair("setgl", SetGlobalLightLevel()),
            Pair("getfaction", GetFactioning()),
            Pair("auth", Terrarum.game.auth),
            Pair("spawnball", SpawnPhysTestBall()),
            Pair("batch", Batch()),
            Pair("settime", SetTime()),
            Pair("gettime", GetTime()),
            Pair("settimedelta", SetTimeDelta()),
            Pair("help", Help()),

            // Test codes
            Pair("bulletintest", SetBulletin()),
            Pair("gsontest", GsonTest())
    )

    fun getCommand(commandName: String): ConsoleCommand {
        return dict[commandName]!!
    }
}
